irs-api/src/main/java/org/eclipse/tractusx/irs/aaswrapper/job/delegate/RelationshipDelegate.java

```diff
@@ -111,30 +112,77 @@ private void processEndpoint(final Endpoint endpoint, final RelationshipAspect r
+            final Tombstone tombstone = createJsonParseSubmodelPayloadTombstone(endpoint, itemId, e);
+            itemContainerBuilder.tombstone(tombstone);
         }
     }
 
+    private Tombstone createNoBpnProvidedTombstone(final Endpoint endpoint, final PartChainIdentificationKey itemId) {
+        final ProcessingError error = createProcessingError(ProcessStep.SUBMODEL_REQUEST, retryCount,
+                "Can't get relationship without a BPN");
+        return createTombstone(endpoint.getProtocolInformation().getHref(), itemId.getGlobalAssetId(), error,
+                itemId.getBpn());
+    }
+
+    private Tombstone createPolicyTombstone(final Endpoint endpoint, final PartChainIdentificationKey itemId,
+            final PolicyException exception) {
+        final Map<String, Object> policy = jsonUtil.asMap(exception.getPolicy());
+        final ProcessingError error = createProcessingError(ProcessStep.USAGE_POLICY_VALIDATION, 0,
+                exception.getMessage());
+        return createTombstone(endpoint.getProtocolInformation().getHref(), itemId.getGlobalAssetId(), error,
+                exception.getBusinessPartnerNumber()).toBuilder().policy(policy).build();
+    }
+
+    private Tombstone createEdcClientExceptionTombstone(final Endpoint endpoint,
+            final PartChainIdentificationKey itemId, final EdcClientException exception) {
+        final ProcessingError error = createProcessingError(ProcessStep.SUBMODEL_REQUEST, 0, exception.getMessage());
+        return createTombstone(endpoint.getProtocolInformation().getHref(), itemId.getGlobalAssetId(), error,
+                itemId.getBpn());
+    }
+
+    private Tombstone createJsonParseSubmodelPayloadTombstone(final Endpoint endpoint,
+            final PartChainIdentificationKey itemId, final JsonParseException exception) {
+        final ProcessingError error = createProcessingError(ProcessStep.SUBMODEL_REQUEST, 0, exception.getMessage());
+        return createTombstone(endpoint.getProtocolInformation().getHref(), itemId.getGlobalAssetId(), error,
+                itemId.getBpn());
+    }
+
+    private ProcessingError createProcessingError(final ProcessStep processStep, final int retryCount,
+            final String exception) {
+        return ProcessingError.builder()
+                              .withProcessStep(processStep)
+                              .withRetryCounterAndLastAttemptNow(retryCount)
+                              .withErrorDetail(exception)
+                              .build();
+    }
+
+    private Tombstone createTombstone(final String endpointURL, final String globalAssetId, final ProcessingError error,
+            final String bpn) {
+        return Tombstone.builder()
+                        .endpointURL(endpointURL)
+                        .catenaXId(globalAssetId)
+                        .processingError(error)
+                        .businessPartnerNumber(bpn)
+                        .build();
+    }
+
```

Comment:
I think it makes sense to replace the overloaded method `from()`  with more descriptive methods. However, we now have a bunch of utility methods to create tombstones in `RelationshipDelegate` instead of `Tombstone`. 
The utility methods should stay in Tombstone IMO.

Reply:
@ds-jhartmann I disagree here. Except for the two createProcessingError and createTombstone methods these are specialized for the relevant catch block (see the hardcoded ProcessStep values). I think we can talk about moving the  reateProcessingError and createTombstone to Tombtone but the reason I did not do this is that I wanted to encourage using the builder instead of adding more and more constructor methods leading to the telescoping constructor (method) anti-pattern.